packages/iTwinUI-react/src/core/utils/components/VirtualScroll.tsx

```diff
@@ -242,6 +214,41 @@ export const useVirtualization = (props: VirtualScrollProps) => {
     };
   }, [visibleChildren.length]);
 
+  const onResize = React.useCallback(
+    ({ height }: DOMRectReadOnly) => {
+      // Initial value returned by resize observer is 0
+      // So wait for the next one
+      if (height > 0) {
+        setIsMounted(true);
+      }
+      setScrollContainerHeight(height);
+
+      updateChildHeight();
+    },
+    [updateChildHeight],
+  );
+  const [resizeRef, resizeObserver] = useResizeObserver(onResize);
+
+  // Find scrollable parent
+  // Needed only on init
+  React.useLayoutEffect(() => {
+    const scrollableParent = getScrollableParent(
+      parentRef.current,
+      parentRef.current?.ownerDocument,
+    );
+    scrollContainer.current = scrollableParent;
+
+    resizeRef(scrollableParent);
+  }, [resizeRef]);
+
+  // Stop watching resize, when virtual scroll is unmounted
+  React.useLayoutEffect(() => {
+    return () => resizeObserver?.disconnect();
+  }, [resizeObserver]);
+
+  // Get child height when children available
+  React.useEffect(() => updateChildHeight(), [updateChildHeight]);
```

Comment:
previously updateChildHeight was layoutEffect. I think you could use the same here.

Reply:
**[DONE]** I changed `useEffect()` to `useLayoutEffect()`. Just curious, what is the difference between the two? I tried looking online but am still not totally sure of the difference

Reply:
Found really good explanation here: https://kentcdodds.com/blog/useeffect-vs-uselayouteffect

Reply:
Very helpful, thanks! The main difference was the synchronous vs asynchronous part. Mentioning that makes everything clearer :smile: